### PR TITLE
feat: Introduce universal settings block for VMs & VMSSs

### PR DESCRIPTION
**Pull request by acelebanski**
## Description

This PR introduces `vmseries_universal` and `scale_sets_universal` variables in examples. They allow to define some properties like `version`, `size`, `bootstrap_options` etc. only once in the tfvars file which apply to all created VMs. It is still possible to override universal settings on a per-VM basis by defining the properties inside a specific VM object in tfvars.

## Motivation and Context

Some users wanted to define some popular properties like PAN-OS version, VM size etc. only once when deploying multiple VMs.

## How Has This Been Tested?

On local workstation by running tf plan and deploying chosen examples.

## Types of changes

- New feature (non-breaking change which adds functionality)

## Checklist

- [x] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes if appropriate.
- [x] All new and existing tests passed.

### CODE REVIEW

**Comment by acelebanski** [comment]
/idempotence paths="examples/common_vmseries examples/common_vmseries_and_autoscale examples/dedicated_vmseries examples/dedicated_vmseries_and_autoscale examples/gwlb_with_vmseries examples/standalone_panorama examples/standalone_vmseries"

File: examples/common_vmseries/variables.tf [hunk 1]
```diff
@@ -605,6 +605,67 @@ variable "bootstrap_storages" {
...
+                          Deployment Guide* as only a few selected sizes are supported.
+  - `bootstrap_options` - (`string`, optional, mutually exclusive with `bootstrap_package`) bootstrap options passed to PAN-OS
+                          when launched for the 1st time, for details see module documentation.
+  - `bootstrap_package` - (`map`, optional, mutually exclusive with `bootstrap_options`) a map defining content of the bootstrap
+                          package. For details and available properties refer to `vmseries` variable.
+  EOF
+  default     = {}
+  type = object({
+    version           = optional(string)
+    custom_id         = optional(string)
```
Comment (alperenkose) [inline examples/common_vmseries/variables.tf:630]: shall we rename it to sth like `custom_image_id` to make it clear what it refers to? Btw does it actually refer to the urn or the really long azure id which contains subscription etc with slashes?

File: examples/common_vmseries/variables.tf [hunk 2]
```diff
@@ -641,8 +703,8 @@ variable "vmseries" {
 
     - `size`              - (`string`, optional, defaults to module default) Azure VM size (type). Consult the *VM-Series
                             Deployment Guide* as only a few selected sizes are supported.
-    - `zone`              - (`string`, optional, defaults to module default) the Availability Zone in which the VM and (if
-                            deployed) public IP addresses will be created.
+    - `zone`              - (`string`, required) the Availability Zone in which the VM and (if deployed) public IP addresses will
```
Comment (alperenkose) [inline examples/common_vmseries/variables.tf:706]: I think `zone` shouldn't be required and vmseries module needs to be fixed to accept zone as optional. In the actual azure vm resource zone is not required.

File: examples/common_vmseries_and_autoscale/README.md [hunk 3]
```diff
@@ -822,6 +823,40 @@ Default value: `&{}`
...
+A map defining common settings for all created VM-Series Scale Sets. 
+  
+It duplicates popular properties from `scale_sets` variable, specifically `scale_sets.image` and 
+`scale_sets.virtual_machine_scale_set` maps. However, if values are set in those maps, they still take precedence over the ones
+set within this variable. As a result, all common properties can be overriden on a per-VMSS basis.
+
+Following properties are supported:
+  
+- `version`           - (`string`, optional) describes the PAN-OS image version from Azure Marketplace.
+- `custom_id`         - (`string`, optional) absolute ID of your own custom PAN-OS image.
```
Comment (alperenkose) [inline examples/common_vmseries_and_autoscale/README.md:837]: thinking the same as for vmseries common custom_id variable.

File: examples/common_vmseries/variables.tf (hunk 2 above)
Comment (acelebanski) [reply examples/common_vmseries/variables.tf:706]: I remember it was a conscious decision to explicitly set `null` as value when you don't want to use availability zones. If we change `zone` to optional, we need to also set the entire `virtual_machine` object as optional, something to think about.

Comment (acelebanski) [reply examples/common_vmseries/variables.tf:706]: We decided to leave it as required on the call.

File: examples/common_vmseries/variables.tf (hunk 1 above)
Comment (acelebanski) [reply examples/common_vmseries/variables.tf:630]: Ok, I'm just getting rid of `custom_id` completely as agreed.

**Review by alperenkose — APPROVED** [review]
👍🏻

**Review by sebastianczech — APPROVED** [review]
👍

**Pull request merged**